src/v/redpanda/admin/services/datalake/datalake.cc

```diff
@@ -0,0 +1,177 @@
+}
+
+proto::admin::lifecycle_state
+to_proto(datalake::coordinator::topic_state::lifecycle_state_t state) {
+    switch (state) {
+    case datalake::coordinator::topic_state::lifecycle_state_t::live:
+        return proto::admin::lifecycle_state::live;
+    case datalake::coordinator::topic_state::lifecycle_state_t::closed:
+        return proto::admin::lifecycle_state::closed;
+    case datalake::coordinator::topic_state::lifecycle_state_t::purged:
+        return proto::admin::lifecycle_state::purged;
+    }
+    return proto::admin::lifecycle_state::unspecified;
+}
+
+proto::admin::topic_state
+to_proto(const datalake::coordinator::topic_state& state) {
+    proto::admin::topic_state pb_state;
+    pb_state.set_revision(state.revision());
+    chunked_hash_map<int32_t, proto::admin::partition_state>
+      pb_partition_states;
+    for (const auto& [pid, pstate] : state.pid_to_pending_files) {
+        pb_partition_states.emplace(pid(), to_proto(pstate));
+    }
+    pb_state.set_partition_states(std::move(pb_partition_states));
+    pb_state.set_lifecycle_state(to_proto(state.lifecycle_state));
+    pb_state.set_total_kafka_bytes_processed(state.total_kafka_bytes_processed);
+    return pb_state;
+}
+} // anonymous namespace
+
+namespace admin {
+
+datalake_service_impl::datalake_service_impl(
+  admin::proxy::client proxy_client,
+  ss::sharded<datalake::coordinator::frontend>* coordinator_fe)
+  : _proxy_client(std::move(proxy_client))
+  , _coordinator_fe(coordinator_fe) {}
+
+ss::future<proto::admin::coordinator_get_state_response>
+datalake_service_impl::coordinator_get_state(
+  serde::pb::rpc::context, proto::admin::coordinator_get_state_request req) {
+    if (!_coordinator_fe->local_is_initialized()) {
+        throw serde::pb::rpc::unavailable_exception(
+          "Datalake coordinator frontend not initialized");
+    }
+
+    // Group topics by coordinator partition.
+    chunked_hash_map<model::partition_id, chunked_vector<model::topic>>
+      topics_by_partition;
+    for (const auto& topic_name : req.get_topics()) {
+        model::topic topic{topic_name};
+        auto partition_opt = _coordinator_fe->local().coordinator_partition(
+          topic);
+        if (!partition_opt.has_value()) {
+            throw serde::pb::rpc::unavailable_exception(
+              fmt::format(
+                "Datalake coordinator couldn't get coordinator partition for "
+                "{}",
+                topic));
+        }
+        topics_by_partition[partition_opt.value()].emplace_back(
+          std::move(topic));
+    }
+
+    // Send out the RPCs.
+    chunked_hash_map<model::topic, datalake::coordinator::topic_state>
+      topic_states;
+    for (const auto& [partition_id, topics] : topics_by_partition) {
+        datalake::coordinator::get_topic_state_request fe_req{
+          partition_id, topics.copy()};
+        auto fe_res = co_await _coordinator_fe->local().get_topic_state(
+          std::move(fe_req));
```

Comment:
q: there are retries built in to the coordinator frontend, right?

Reply:
Nah, only routing to the leader is done by the frontend. I'd considered adding retries in this PR, but opted to keep this PR slim